packages/model-fragments/lib/fragments/model.js

```diff
@@ -241,5 +135,29 @@ function getActualFragmentType(declaredType, options, data) {
   return actualType || declaredType;
 }
 
+// Returns the internal model for the given record/fragment
+export function internalModelFor(record) {
+  var internalModel = record._internalModel;
+
+  // Ensure the internal model has a fragments hash, since we can't override the
+  // constructor function anymore
+  if (!internalModel._fragments) {
+    internalModel._fragments = create(null);
+  }
+
+  return internalModel;
+}
```

Comment:
@igorT Overall, I was unhappy with how heavily this relies on `_internalModel`, because it seemed like a step backward. In retrospect, I think it's more of a side-step, since MF has always made use of private APIs like `._data`. I'd love to see it move away from touching model internals, but there are certain things that are unavoidable, i.e. storing state on the model instance.

This method is another `InternalModel` workaround, since there's no longer an `init` hook I can take advantage of to ensure the `._fragments` hash exists.